src/sonic-bgpcfgd/tests/test_static_rt_cleanup.py

```diff
@@ -0,0 +1,173 @@
+import unittest
+from unittest.mock import MagicMock, patch, call
+
+from bgpcfgd.directory import Directory
+from bgpcfgd.template import TemplateFabric
+from bgpcfgd.managers_static_rt import StaticRouteMgr, IpNextHopSet, IpNextHop
+from bgpcfgd.main import cleanup_static_routes_on_exit
+from swsscommon import swsscommon
+import socket
+
+
+class TestStaticRouteCleanup(unittest.TestCase):
+    
+    def setUp(self):
+        """Set up test fixtures before each test method."""
+        self.cfg_mgr = MagicMock()
+        self.directory = Directory()
+        
+        self.common_objs = {
+            "directory": self.directory,
+            "cfg_mgr": self.cfg_mgr,
+            "tf": TemplateFabric(),
+            "constants": {},
+        }
+        
+        # Create StaticRouteMgr instances for testing
+        self.config_mgr = StaticRouteMgr(self.common_objs, "CONFIG_DB", "STATIC_ROUTE")
+        self.appl_mgr = StaticRouteMgr(self.common_objs, "APPL_DB", "STATIC_ROUTE")
+        
+        # Setup BGP ASN for redistribution tests
+        self.directory.put("CONFIG_DB", swsscommon.CFG_DEVICE_METADATA_TABLE_NAME, "localhost", {"bgp_asn": "65100"})
+    
+    def test_cleanup_on_exit_no_routes(self):
+        """Test cleanup_on_exit when no routes are cached."""
+        # Clear any routes
+        self.config_mgr.static_routes = {}
+        
+        # Call cleanup_on_exit
+        result = self.config_mgr.cleanup_on_exit()
+        
+        # Should return None (void method) and not call cfg_mgr
+        self.assertIsNone(result)
```

Comment:
do we need this and the same below? the _cleanup_on_exit()_ newer returns anything